keps/sig-scheduling/5007-device-attach-before-pod-scheduled/README.md

```diff
@@ -0,0 +1,1117 @@
+
+Regarding collaboration with the Cluster Autoscaler, using node nomination can address the issue.
+This issue needs to be resolved before the beta is released.
+
+**The in-flight events cache may grow too large when waiting in PreBind.**
+
+To address the PreBind concern, the solution is to modify the scheduling framework to flush the in-flight events cache before PreBind.
+This prevents issues in the scheduling queue caused by keeping pods at PreBind for an extended period.
+This issue will be addressed separately as outlined in kubernetes/kubernetes#129967.
+This issue needs to be resolved before the beta is released.
+
+## Design Details
+
+### DRA Scheduler Plugin Design Overview
+
+This document outlines the design of the DRA Scheduler Plugin, focusing on the handling of fabric devices.
+Key additions include new attributes for device identification, enhancements to `AllocatedDeviceStatus`, and the process for handling `ResourceSlices` upon attachment failure.
+The composable controller design is also discussed, emphasizing efficient utilization of fabric devices.
+
+![proposal](proposal.jpg)
+
+#### BasicDevice Enhancements
+
+To indicate whether a device is a fabric device, fields are added to the `Basic` within `Device`.
+These fields will be used by the controller that exposes the `ResourceSlice` to notify whether the device is a fabric device.
+
+```go
+// Device represents one individual hardware instance that can be selected based
+// on its attributes. Besides the name, exactly one field must be set.
+type Device struct {
+    // Name is unique identifier among all devices managed by
+    // the driver in the pool. It must be a DNS label.
+    //
+    // +required
+    Name string
+
+    // Basic defines one device instance.
+    //
+    // +optional
+    // +oneOf=deviceType
+    Basic *BasicDevice
+}
+
+// BasicDevice represents a basic device instance.
+type BasicDevice struct {
+    // Attributes defines the set of attributes for this device.
+    // The name of each attribute must be unique in that set.
+    //
+    // The maximum number of attributes and capacities combined is 32.
+    //
+    // +optional
+    Attributes map[QualifiedName]DeviceAttribute
+
+    // BindingGates defines the gates for binding.
+    //
+    // +optional
+    BindingGates []string
+
+    // BindingFailureGates defines the gates for binding failure.
+    //
+    // +optional
+    BindingFailureGates []string
+
+    // UsageRestrictedToNode indicates if the usage of an allocation involving this device
+    // has to be limited to exactly the node that was chosen when allocating the claim.
+    //
+    // +optional
+    UsageRestrictedToNode bool
+
+    // BindingTimeout indicates the prepare timeout period(minute).
+    // If the timeout period is exceeded, the scheduler clears the allocation in the ResourceClaim and reschedules the Pod.
+    //
+    // +optional
+    BindingTimeout int
```

Comment:
To avoid ambiguity, other durations are explicitly called `Seconds`, so `BindingTimeoutSeconds`.

Or use `BindingTimeout *metav1.Duration`. Probably better.